### DISCUSSION TITLE
Running in production environments (Docker/Kubernetes)

### DISCUSSION THREAD

**Discussion by bragafelipe** [taskforcesh/bullmq#665] [Q&A] [answered]
I want to know the recommend approach to run bullmq with docker for a production environment (k8s). I´m running the getting started example from https://docs.bullmq.io/. 

Breaking down the question:

1. How should I run it within docker? Directly with node using "node index.js"? Should I use a process manager?

2. How would I configure liveness and readiness? Should I use a express to have a /health with some logic to validate basic things (e.g. database connectivity)?

Thanks in advance.

**Comment by manast** [comment, marked as answer]
> * How should I run it within docker? Directly with node using "node index.js"? Should I use a process manager?
> *

I think this question is not specific for BullMQ but to any NodeJs application. Although process managers like pm2 will recommend you to use them even in docker, I personally do not do that and rely on dockers own auto restart policies and health check mechanisms. 


> 2\. How would I configure liveness and readiness? Should I use a express to have a /health with some logic to validate basic things (e.g. database connectivity)?

Same thing here really. There is nothing fundamentally different with a BullMQ worker than any other NodeJs server application, I recommend you to read  elsewhere what others recommend since this is quite subjective and there are many pros and cons for the different alternatives.

**Reply by bragafelipe** [reply]
I understand what you meant, at first I was looking at bull not only as a library but also as a framework. Anyway makes a lot of sense, I will research as a you suggested. Thanks for taking the time to reply.
